File: BaseMvp/src/main/java/com/base/applemvp/annotations/CreatePresenterAnnotation.java
package com.base.applemvp.annotations;



import com.base.applemvp.common.BasePresenter;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * @author applehsp
 */
@Retention(RUNTIME)
@Target(FIELD)
public @interface CreatePresenterAnnotation {
    Class<? extends BasePresenter> value();
}

File: BaseMvp/src/main/java/com/base/applemvp/annotations/CreatePresenterAnnotation.kt
package com.base.applemvp.annotations

import com.base.applemvp.common.BasePresenter
import com.base.applemvp.common.IBaseView
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import kotlin.reflect.KClass

/**
 * @author applehsp
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class CreatePresenterAnnotation(val value: KClass<out BasePresenter<out IBaseView>>)